From 9fa099d089186a9b441586a0257463acb95b0412 Mon Sep 17 00:00:00 2001
Message-Id: <9fa099d089186a9b441586a0257463acb95b0412.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Shobhit Kumar <shobhit.kumar@intel.com>
Date: Wed, 1 Oct 2014 16:33:38 +0530
Subject: [PATCH 69/75] REVERTME [VPG]: adf/intel/vlv: Disable switching VSYNC
 event off

Looks like usermode is calling set event IOCTLS for VSYNC ON/OFF
regulalry which are not correctly paired or delay between OFF to ON
which is causing vsync timeouts during the preiod which VSYNC is off

With this WA, vsync are always kept on and we do not see any timeouts.
Seems issue need to be fixed in kernel. Will rework on this.

For: GMINL-3547
Change-Id: Id22f97bae8ef852985b5d44fc41e0d633fcfdaa7
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
---
 .../video/adf/intel/core/common/dsi/gen_dsi_pipe.c |    9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c b/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
index c1a0f1f..0a9b8e6 100644
--- a/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
+++ b/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
@@ -280,8 +280,13 @@ int dsi_set_event(struct intel_pipe *pipe, u16 event, bool enabled)
 
 	if (enabled)
 		pipestat = 0;
-	else
-		pipestat = 0xFFFFFFFF;
+	else {
+		if (event == INTEL_PIPE_EVENT_VSYNC) {
+			pr_debug("ADF: %s: Not allowing VSYNC OFF\n", __func__);
+			return 0;
+		} else
+			pipestat = 0xFFFFFFFF;
+	}
 
 	switch (event) {
 	case INTEL_PIPE_EVENT_SPRITE2_FLIP:
-- 
1.7.9.5

